src/net/sourceforge/kolmafia/Modifiers.java

```diff
@@ -833,10 +834,25 @@ public static synchronized void availableSkillsChanged() {
   }
 
   public void addExpression(Indexed<DoubleModifier, ModifierExpression> entry) {
+    int index = -1;
+
     if (this.expressions == null) {
       this.expressions = new ArrayList<>();
+    } else {
+      for (int i = 0; i < this.expressions.size(); i++) {
+        Indexed<DoubleModifier, ModifierExpression> e = this.expressions.get(i);
+        if (e != null && e.index == entry.index) {
+          index = i;
+          break;
+        }
+      }
+    }
+
+    if (index < 0) {
+      this.expressions.add(entry);
+    } else {
+      this.expressions.get(index).value.combine(entry.value, '+');
```

Comment:
Is this literally the only place you call the new "combine" method on an Expression?
So, your list of operators to "combine" only matters if the operator is '+'?

Reply:
Yes, but given the `combine()` method edits the "program code" directly, I didn't want to let people accidentally corrupt the "stack" by using a unary 

I can just swap it to an `add(Expression other)` method - just thought given there's the logic for remapping the literals etc might as well keep it generic!